src/Analyzer.Reports/IReportWriter.cs

```diff
@@ -18,6 +18,7 @@ public interface IReportWriter : IDisposable
         /// <param name="evaluations">Evaluation list</param>
         /// <param name="templateFile">Template file to be analyzed</param>
         /// <param name="parametersFile">The parameter file to use when parsing the specified ARM template.</param>
-        void WriteResults(IEnumerable<Types.IEvaluation> evaluations, IFileInfo templateFile, IFileInfo parametersFile = null);
+        /// <param name="configurationsFile">The configurations file to use when running the analyzer</param>
+        void WriteResults(IEnumerable<Types.IEvaluation> evaluations, IFileInfo templateFile, IFileInfo parametersFile = null, IFileInfo configurationsFile = null);
```

Comment:
Since the configuration file applies to the entire run, it's better to pass that information in the constructor of a report writer instead of when writing results, since the config will apply globally and not just to a set of results.